src/authentication.ts

```diff
@@ -3,7 +3,7 @@ import bsv from 'bsv'
 import { RandomHex, Hash } from './utils'
 import { deriveHDPrivateChildKeyFromHex } from './utils/keys'
 import { signMessage } from './utils/sign'
-import logger from './logger'
+import { logger } from './logger'
```

Comment:
I'm not sure, but with this approach users still can't configure/add a custom logger

Reply:
I guess that the logger should be passed as an option/argument to the client and then used.

Something like that:

```typescript
interface Logger {
  debug(msg: string, ...args: any[]): void;
  info(msg: string, ...args: any[]): void;
  warn(msg: string, ...args: any[]): void;
  error(msg: string, ...args: any[]): void;
}

interface SimpleLoggerConfig {
  level: "debug" | "info" | "warn" | "error";
}

type LoggerConfig = Logger | SimpleLoggerConfig;

const isSimpleLoggerConfig = (config: LoggerConfig): config is SimpleLoggerConfig => {
  return (config as SimpleLoggerConfig).level !== undefined;
}

const nop = () => {}

export class SpvWalletClient {
  // (...)
  logger: Logger

  constructor(/* ... */ loggerConfig?: LoggerConfig) {
    // (...)

    if(loggerConfig == null) {
      this.logger = console
    }else if(isSimpleLoggerConfig(loggerConfig)) {
      const { level } = loggerConfig
      this.logger = {
        debug: level === 'debug' ? console.debug : nop,
        info: level === 'debug' || level === 'info' ? console.info : nop,
        warn: level === 'debug' || level === 'info' || level === 'warn' ? console.warn : nop,
        error: console.error,
      }
    } else {
      this.logger = loggerConfig
    }
    // (...)
}